app/domain/authentication/handler/oidc_authentication_handler.rb

```diff
@@ -59,6 +59,21 @@ def type
         return 'oidc'
       end
 
+
+      def generate_token(account, identity)
+        @token_factory.signed_token(
+          account: account,
+          username: conjur_identity(account: account, id: identity)
+        )
+      end
+
+      def conjur_identity(account:, id:)
+        annotation = Annotation.fetch_annotation(account: account, value: id).first
+        return id unless annotation
```

Comment:
This definitely needs a good comment describing what's going on here.

This is still giving precedence to annotations over role ID's, correct?

Reply:
from what i understood yes. i asked while talking about this feature with jason if we wanted to default to role_id, then use annotation if it didn't match and Jason suggested the other way around. This wasn't about a specific section of code tho but more of a high level discussion

Reply:
Hey @jvanderhoof , can you give any additional context to why we would give precedence for annotation values over resource IDs when resolving OIDC authentications?

The risk I'm considering is that this allows a host to set an annotation and impersonate an existing role with the host ID used for the mapping. Also, it's generally been an implied design principle that annotations are a weaker security control than IDs or variables. So taking an annotation as the higher precedent over the ID doesn't seem to jive with that either.

I'm probably missing some background or context that's important here, though.

CC: @andytinkham

Reply:
@micahlee, as mentioned above, with annotations, I'm hoping to provide a path for getting users with email addresses (common OIDC pattern) mapped to Conjur users (where we use the "unfortunate" `@` to delineate policy scope for a user - `my.user@my-policy`).

Annotations have emerged as a common pattern for attributing external ID to internal ID. I agree it's less secure. We should absolutely keep pushing on this to understand how best to manage to "Users in root" requirement and tie internal/external mapping.